specs/BUNDLE_SPEC.md

```diff
@@ -0,0 +1,271 @@
+# Cosign Bundle Specification
+
+This document aims to describe how `cosign` attaches Sigstore attestation
+[bundles](https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_bundle.proto)
+to container images.
+
+The goal is to specify the behavior well enough to promote other implementations
+and enable interoperability. Attestations attached with `cosign` should be
+retrievable in other tools, and vice-versa.
+
+This document focuses on the layout of attestations within an
+[OCI Image Manifest V1.1](https://github.com/opencontainers/image-spec/blob/v1.1.0/manifest.md)
+object.
+
+This document makes no assumptions about the contents of the Sigstore bundle.
+Any attestation which can be represented as a Sigstore bundle (message
+signatures, DSSE-wrapped in-toto statements, etc) can be attached to a container
+image stored in an OCI registry.
+
+Multiple Attestations may be "attached" to one image.
+
+Attestations attached to a container image are generally assumed to refer to
+that image in some way.
+
+## Storage
+
+The approach for storing Sigstore bundles in an OCI registry follows the
+[guidelines for artifact usage](https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage)
+in the OCI
+[image spec](https://github.com/opencontainers/image-spec/blob/main/README.md).
+
+### Publishing
+
+First, the bundle itself is stored in its JSON-serialized form as a blob in the
+registry:
+
+```
+POST /v2/foo/blobs/uploads/?digest=cafed00d...
+Content-Type: application/octet-stream
+
+{"mediaType":"application/vnd.dev.sigstore.bundle+json;version=0.2", ...}
+```
+
+In this example “foo” is the name of the repository within the registry to which
+the artifact is being uploaded. The digest included as part of the POST is the
+hex-encoded SHA-256 digest of the raw bytes of the bundle itself.
+
+Once the blob has been created, the next step is to create a manifest that
+associates the bundle blob with the image it describes:
+
+```
+PUT /v2/foo/manifests/sha256:badf00d...
+Content-Type: application/vnd.oci.image.manifest.v1+json
+
+{
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "schemaVersion": 2,
+  "artifactType": "application/vnd.dev.sigstore.bundle+json;version=0.2",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa3...",
+    "size": 2
+  },
+  "layers": [
+    {
+      "digest": "sha256:cafed00d...",
+      "mediaType": "application/vnd.dev.sigstore.bundle+json;version=0.2",
```

Comment:
Similar issue to above, with the parameters on the media type likely to see support issues.